test/powershell/Modules/Microsoft.PowerShell.Core/Enter-PSHostProcess.Tests.ps1

```diff
@@ -2,54 +2,160 @@
+
+            "Enter-PSHostProcess -DebugPipeName $pipeName -ErrorAction Stop
+            `$pid
+            Exit-PSHostProcess" | pwsh -c - | Should -Be $pwsh.Id
+        }
+
+        It "Can enter, exit, and re-enter using DebugPipeName" {
+            Wait-UntilTrue { Test-Path $pipePath }
 
-    It "Can enter using NamedPipeConnectionInfo" {
-        $npInfo = [System.Management.Automation.Runspaces.NamedPipeConnectionInfo]::new($pwsh.Id)
-        $rs = [runspacefactory]::CreateRunspace($npInfo)
-        $rs.Open()
-        $ps = [powershell]::Create()
-        $ps.Runspace = $rs
-        $ps.AddScript('$pid').Invoke() | Should -Be $pwsh.Id
-        $rs.Dispose()
+            "Enter-PSHostProcess -DebugPipeName $pipeName -ErrorAction Stop
+            `$pid
+            Exit-PSHostProcess" | pwsh -c - | Should -Be $pwsh.Id
+
+            "Enter-PSHostProcess -DebugPipeName $pipeName -ErrorAction Stop
+            `$pid
+            Exit-PSHostProcess" | pwsh -c - | Should -Be $pwsh.Id
+        }
+
+        It "Should throw if DebugPipeName does not exist" {
+            Wait-UntilTrue { Test-Path $pipePath }
+
+            { Enter-PSHostProcess -DebugPipeName badpipename } | Should -Throw -ExpectedMessage "No named pipe was found with DebugPipeName: badpipename."
+        }
+
+        It "Should throw if a pipe name is too long on Linux or macOS" {
+            $longPipeName = "DoggoipsumwaggywagssmolborkingdoggowithalongsnootforpatsdoingmeafrightenporgoYapperporgolongwatershoobcloudsbigolpupperlengthboy"
+
+            if (!$IsWindows) {
+                pwsh -debugpipename $longPipeName | Should -BeLike "Cannot process the command because -DebugPipeName specified is too long.*"
+            } else {
+                pwsh -debugpipename $longPipeName
+                $LASTEXITCODE | Should -Be 0
+            }
+        }
+
+        It "Should be able to change the pipename via the API" {
+            Wait-UntilTrue { Test-Path $pipePath }
+
+            $pipeName1 = "myTestPipe1"
+            $pipeName2 = "myTestPipe2"
+
+            [System.Management.Automation.Remoting.RemoteSessionNamedPipeServer]::CreateCustomNamedPipeServer($pipeName1)
+            Get-PipePath -PipeName $pipeName1 | Test-Path | Should -BeTrue
+
+            # The second call to this method would override the first named pipe.
+            [System.Management.Automation.Remoting.RemoteSessionNamedPipeServer]::CreateCustomNamedPipeServer($pipeName2)
+            Get-PipePath -PipeName $pipeName2 | Test-Path | Should -BeTrue
+
+            # Previous pipe should have been cleaned up.
+            Get-PipePath -PipeName $pipeName1 | Test-Path | Should -BeFalse
+        }
+
+        It "Should throw if a pipe name is too long on Linux or macOS" {
+            $longPipeName = "DoggoipsumwaggywagssmolborkingdoggowithalongsnootforpatsdoingmeafrightenporgoYapperporgolongwatershoobcloudsbigolpupperlengthboy"
+
+            if (!$IsWindows) {
+                try {
```

Comment:
Is there a reason you couldn't use the pattern:

{ ...scriptblock... | | Should -Throw -ErrorId "FullyQualifiedErrorId to validate"

Reply:
Ah! I was looking for that. Fixed.